scenes/game/game.gd

```diff
@@ -2499,6 +2501,22 @@ func _on_force_die_result_event(event_data):
 		# Nothing for opponent.
 		pass
 
+func _on_life_damage_dealt(event_data):
+	var active_player = get_player(event_data["event_player_id"])
+	var target_player = get_player(event_data["target_player"])
+	var source_card_id = event_data["source_card_id"]
+	var life_lost = event_data["life_lost"]
+
+	game_log.add_to_log(
+		GameLog.GameLogLine.Detail,
+		"%s [CARD]%s[/CARD] deals %s life damage to %s" % [
+			active_player.get_name_decorated(),
+			_get_card_definition_id(source_card_id),
+			life_lost,
+			target_player.get_name_decorated()
+		])
+	_play_popup_message("Deal %s life damage" % life_lost)
```

Comment:
So you kind of need to do whatever happens in _process_downed_holomem in regards to life lost, right?
This also needs tr() if you want a message here.